presto-main/src/main/java/io/prestosql/execution/buffer/LazyOutputBuffer.java

```diff
@@ -99,21 +100,23 @@ public double getUtilization()
 
         // until output buffer is initialized, it is "full"
         if (outputBuffer == null) {
-            return 1.0;
+            return OptionalDouble.empty();
         }
         return outputBuffer.getUtilization();
     }
 
     @Override
-    public boolean isOverutilized()
+    public BufferState getState()
     {
         OutputBuffer outputBuffer;
         synchronized (this) {
             outputBuffer = delegate;
         }
 
-        // until output buffer is initialized, readers cannot enqueue and thus cannot be blocked
-        return (outputBuffer != null) && outputBuffer.isOverutilized();
+        if (outputBuffer == null) {
```

Comment:
what is state of lazy buffer, `OPEN`?

Reply:
Yes, lazy buffer also starts with and `OPEN` state.